src/common/bit_str.h

```diff
@@ -17,6 +17,8 @@
 #include <cstdint>
 #include <iosfwd>
 #include <functional>
+#include <cstdint>
+#include <ostream>
```

Comment:
what was the error message? if we need a complete type definition of `std::ostream`. we could instead include `<ostream>` in `bit_str.cc` and just have `<iosfwd>` in the header file.